hoodie-spark/src/main/scala/com/uber/hoodie/DefaultSource.scala

```diff
@@ -177,93 +178,106 @@ class DefaultSource extends RelationProvider
+        throw new HoodieException(s"basePath ${basePath} already exists.")
+      }
+      if (mode == SaveMode.Ignore && exists) {
+        log.warn(s" basePath ${basePath} already exists. Ignoring & not performing actual writes.")
+        return createRelation(sqlContext, parameters, df.schema)
+      }
+      if (mode == SaveMode.Overwrite && exists) {
+        log.warn(s" basePath ${basePath} already exists. Deleting existing data & overwriting with new data.")
+        fs.delete(basePath, true)
+        exists = false
+      }
 
-    // Create the dataset if not present (APPEND mode)
-    if (!exists) {
-      val properties = new Properties();
-      properties.put(HoodieTableConfig.HOODIE_TABLE_NAME_PROP_NAME, tblName.get);
-      properties.put(HoodieTableConfig.HOODIE_TABLE_TYPE_PROP_NAME, storageType);
-      properties.put(HoodieTableConfig.HOODIE_ARCHIVELOG_FOLDER_PROP_NAME, "archived");
-      HoodieTableMetaClient.initializePathAsHoodieDataset(sparkContext.hadoopConfiguration, path.get, properties);
-    }
+      // Create the dataset if not present (APPEND mode)
+      if (!exists) {
+        val properties = new Properties();
+        properties.put(HoodieTableConfig.HOODIE_TABLE_NAME_PROP_NAME, tblName.get);
+        properties.put(HoodieTableConfig.HOODIE_TABLE_TYPE_PROP_NAME, storageType);
+        properties.put(HoodieTableConfig.HOODIE_ARCHIVELOG_FOLDER_PROP_NAME, "archived");
+        HoodieTableMetaClient.initializePathAsHoodieDataset(sparkContext.hadoopConfiguration, path.get, properties);
+      }
 
-    // Create a HoodieWriteClient & issue the write.
-    val jsc = new JavaSparkContext(sparkContext);
-    val client = DataSourceUtils.createHoodieClient(jsc,
-      schema.toString,
-      path.get,
-      tblName.get,
-      mapAsJavaMap(parameters)
-    )
-    val commitTime = client.startCommit();
+      // Create a HoodieWriteClient & issue the write.
+      val jsc = new JavaSparkContext(sparkContext);
+      val client = DataSourceUtils.createHoodieClient(jsc,
+        schema.toString,
+        path.get,
+        tblName.get,
+        mapAsJavaMap(parameters)
+      )
+      val commitTime = client.startCommit();
 
-    val writeStatuses = DataSourceUtils.doWriteOperation(client, hoodieRecords, commitTime, operation)
-    // Check for errors and commit the write.
-    val errorCount = writeStatuses.rdd.filter(ws => ws.hasErrors).count()
-    if (errorCount == 0) {
-      log.info("No errors. Proceeding to commit the write.");
-      val metaMap = parameters.filter(kv =>
-        kv._1.startsWith(parameters(COMMIT_METADATA_KEYPREFIX_OPT_KEY)))
-      val success = if (metaMap.isEmpty) {
-        client.commit(commitTime, writeStatuses)
-      } else {
-        client.commit(commitTime, writeStatuses,
-          Optional.of(new util.HashMap[String, String](mapAsJavaMap(metaMap))))
-      }
+      val writeStatuses = DataSourceUtils.doWriteOperation(client, hoodieRecords, commitTime, operation)
```

Comment:
It does not make sense to allow upsert() operation if INSERT_DROP_DUPS_OPT_KEY is set. Can you auto-correct the operation to "insert" if OPERATION_OPT_KEY is not set or if set to "upsert" wrongly.

Reply:
will do

Reply:
Just confirimg.. any code changes where made in the blocks of code that was moved? Does nt seem like it..

Reply:
not intending to, will double check against new master

Reply:
@vinothchandar double checked after rebasing with new master, no code changes in the blocks that was moved.